src/cli/migrate.command.ts

```diff
@@ -0,0 +1,154 @@
+          ),
+        };
+      case 'orders':
+        return {
+          type: 'orders',
+          callback: this.orderImportService.migrateOrders.bind(
+            this.orderImportService,
+          ),
+        };
+      case 'customers':
+        return {
+          type: 'customers',
+          callback: this.customerImportService.migrateCustomers.bind(
+            this.customerImportService,
+          ),
+        };
+    }
+    return null;
+  }
+
+  @Option({
+    flags: '--days [number]',
+    description: 'Sync from last X days',
+  })
+  parseDays(val: string): string {
+    const date = new Date();
+    const matched = val.match(/[0-9]{1,}/g);
+    date.setDate(date.getDate() - parseInt(matched[0]));
+    return date.toJSON();
+  }
+
+  @Option({
+    flags: '--hours [number]',
+    description: 'Sync from last X hours',
+  })
+  parseHours(val: string): string {
+    const date = new Date();
+    const matched = val.match(/[0-9]{1,}/g);
+    date.setHours(date.getHours() - parseInt(matched[0]));
+    return date.toJSON();
+  }
+  @Option({
+    flags: '--ms [number]',
+    description: 'Sync from last X milliseconds',
+  })
+  parseMs(val: string): string {
+    const date = new Date();
+    const matched = val.match(/[0-9]{1,}/g);
+    date.setMilliseconds(date.getMilliseconds() - parseInt(matched[0]));
+    return date.toJSON();
+  }
+
+  @Option({
+    flags: '--date [string]',
+    description: `Sync from specific date. Format: 'YYYY-MM-DD'`,
+  })
+  parseDate(val: string): string {
+    const date = new Date();
+    const matched = val.match(/[0-9]{4}-[0-9]{2}-[0-9]{2}/g);
+    date.setTime(Date.parse(matched[0]));
+    return date.toJSON();
+  }
+
+  @Option({
+    flags: '--longdate [string]',
+    description: `Sync from specific date and time. Format: 'YYYY-MM-DDTHH:MM:SS'`,
+  })
+  parseLongDate(val: string): string {
+    const date = new Date();
+    const matched = val.match(
+      /[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}/g,
+    );
+    date.setTime(Date.parse(matched[0]));
+    return date.toJSON();
+  }
+
+  async run(
+    passedParams: string[],
+    options: MigrateCommandOptions,
+  ): Promise<void> {
+    if (!options.type) {
+      this.logger.warn(
+        `You didn't pass --type argument or it's value is not correct. Possible values are : 'products', 'orders', 'customers'`,
+      );
+      return;
+    }
+
+    const optionKeyOtherThanType = Object.keys(options).find(
+      (key) => key != 'type',
+    );
+    const fetchTime = options[optionKeyOtherThanType]
+      ? options[optionKeyOtherThanType]
+      : null;
+
+    const spinnerCouponsTypes = loadingCli(
+      fetchTime
+        ? `Attempt to migrate ${options.type.type} from ${fetchTime}`
+        : `Attempt to migrate ${options.type.type}`,
+    ).start();
+
+    const result = await options.type.callback(fetchTime);
+
+    if (result.success) {
+      spinnerCouponsTypes.succeed(`${options.type.type} successfully migrated`);
```

Comment:
when you see something like `options.type.type` then usually it means that naming can be improved. For example, if  in `parseType` function you change returned object from
```
{type, callback}
```
to
```
{name, callback}
```
then here you will have 

```
options.type.name
```

what makes much more sense.